lib/core/src/display/shape/text/text_field/content.rs

```diff
@@ -260,17 +290,48 @@ impl TextFieldContent {
         let index    = is_valid.and_option_from(|| Some(index as usize));
         index.map(move |i| self.line(i))
     }
+
+    /// Converts location in this text represented by `row:column` pair to absolute char's position
+    /// from document begin.
+    pub fn convert_location_to_char_index(&mut self, location:TextLocation) -> usize {
+        if self.line_offsets.is_empty() {
+            self.line_offsets.push(0);
+        }
+        if self.line_offsets.len() <= location.line {
+            let required_len           = location.line + 1;
+            let mut current_char_index = *self.line_offsets.last().unwrap();
+            for line in self.lines.iter().take(required_len).skip(self.line_offsets.len()-1) {
+                current_char_index += line.len() + 1;
+                self.line_offsets.push(current_char_index);
+            }
+        }
+        self.line_offsets[location.line] + location.column
+    }
+
+    /// Converts range of locations in this text represented by `row:column` pair to absolute
+    /// char's position from document begin.
+    pub fn convert_location_range_to_char_index(&mut self, range:&Range<TextLocation>)
+    -> Range<usize> {
```

Comment:
Perhaps `Index` from `data::text` could replace here (and in some other API usage) `usize`.